api/django-fileupload/django_fileupload/views.py

```diff
@@ -52,6 +54,10 @@ def create(self, request, *args, **kwargs):
             files = request.FILES.getlist("files")
             if self.verify_file_count(request, len(files)):
                 for file_position, file in enumerate(files):
+
+                    if self.max_file_size and file.size > self.max_file_size:
+                        raise ValidationError(_(f"File size exceeds the maximum allowed size of {self.max_file_size / (1024 ** 2)} MB."))
```

Comment:
This should be done at the nginx level, otherwise it is possible to overload the system with files that are too large. See: https://github.com/ETH-NEXUS/stoffels-cybathlon-2022/blob/main/ws/nginx/conf.d/custom/answer_attachment.nginx

Reply:
The reason for having the max file size defined here is because we need to have it configurable per project workflow as part of the bmip project. In that case an Nginx based solution would not offer the flexibility needed. Could you provide more details about what you mean with "could possibly overload the system"?

Reply:
When a user uploads a very large file, the file content is loaded completely into memory. This memory allocation can put a strain on the system. Only then can one check the upload size in Django. That is why setting upload size limits in nginx is important. A nice feature would be stream-based uploads, but Django does not support this (yet).

Reply:
Not quite right. Django has a memory AND a file-based upload handler. See also: https://github.com/ETH-NEXUS/nexus-app-stack-contrib/blob/main/api/django-common/django_common/uploadhandlers.py

Reply:
Added the custom file upload handlers